### PR TITLE
Add sanity check in applyAllowListUpdates

### PR DESCRIPTION
**Pull request by RayXpub**
## Motivation
Currently, in the event of a misconfig where a destination chain's `AllowListConfigArgs` `allowListEnabled` is set to false while providing addresses to add to the allow list, the config will silently be skipped.

## Solution
Add a sanity check to detect `!allowListConfigArgs.allowListEnabled && `allowListConfigArgs.addedAllowlistedSenders.length > 0`

### CODE REVIEW

File: contracts/src/v0.8/ccip/test/onRamp/OnRamp.t.sol [hunk 1]
```diff
@@ -854,7 +854,7 @@ contract OnRamp_applyDestChainConfigUpdates is OnRampSetup {
   }
 }
 
-contract OnRamp_allowListConfigUpdates is OnRampSetup {
+contract OnRamp_applyAllowListUpdates is OnRampSetup {
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/test/onRamp/OnRamp.t.sol:857]: nit: naming for test functions should be updated to match

File: contracts/src/v0.8/ccip/onRamp/OnRamp.sol [hunk 2]
```diff
@@ -420,6 +420,8 @@ contract OnRamp is IEVM2AnyOnRampClient, ITypeAndVersion, OwnerIsCreator {
         if (allowListConfigArgs.addedAllowlistedSenders.length > 0) {
           emit AllowListSendersAdded(allowListConfigArgs.destChainSelector, allowListConfigArgs.addedAllowlistedSenders);
         }
+      } else if (allowListConfigArgs.addedAllowlistedSenders.length > 0) {
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/onRamp/OnRamp.sol]: nit: probably more efficient (& readable) to de-dupe this conditional:


```solidity
if (allowListConfigArgs.addedAllowlistedSenders.length > 0) {
  if (allowlistConfigArgs.allowlistEnabled) {
     // run loop
    emit AllowListSendersAdded(...);
  } else {
     revert ();
  }
}
```

**Review by elatoskinas — APPROVED** [review]

**Review by RyanRHall — APPROVED** [review]

File: contracts/src/v0.8/ccip/test/onRamp/OnRamp.t.sol (hunk 1 above)
Comment (RayXpub) [reply contracts/src/v0.8/ccip/test/onRamp/OnRamp.t.sol:857]: Renamed

File: contracts/src/v0.8/ccip/onRamp/OnRamp.sol (hunk 2 above)
Comment (RayXpub) [reply contracts/src/v0.8/ccip/onRamp/OnRamp.sol]: Good call, updated.

File: contracts/src/v0.8/ccip/onRamp/OnRamp.sol [hunk 3]
```diff
@@ -408,20 +408,24 @@ contract OnRamp is IEVM2AnyOnRampClient, ITypeAndVersion, OwnerIsCreator {
...
+              revert InvalidAllowListRequest(allowListConfigArgs.destChainSelector);
+            }
+            destChainConfig.allowedSendersList.add(toAdd);
           }
-          destChainConfig.allowedSendersList.add(toAdd);
-        }
 
-        if (allowListConfigArgs.addedAllowlistedSenders.length > 0) {
-          emit AllowListSendersAdded(allowListConfigArgs.destChainSelector, allowListConfigArgs.addedAllowlistedSenders);
+          if (allowListConfigArgs.addedAllowlistedSenders.length > 0) {
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/onRamp/OnRamp.sol]: This conditional can be removed (since it's done in L411)

Comment (RayXpub) [reply contracts/src/v0.8/ccip/onRamp/OnRamp.sol]: Removed

**Pull request merged**